app/Domains/Tags/Actions/Queries/GetTags.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domains\Tags\Actions\Queries;
+
+use App\Domains\Tags\Tag;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Support\Facades\Response;
+use Lorisleiva\Actions\ActionRequest;
+use Lorisleiva\Actions\Concerns\AsAction;
+
+
+class GetTags
+{
+    use AsAction;
+
+    public function authorize(ActionRequest $request): Response
+    {
+        $user = $request->user();
+        
+        if ($user->has_general_access) {
+            return Response::allow();
+        }
+
+        return Response::deny('You are unauthorized to perform this action');
+    }
+
+    public function handle(?int $per_page = 10, ?string $search_term = ''): Collection | LengthAwarePaginator
+    {
+        $query = Tag::query();
+
+        if ($search_term) {
+            $query
+                ->where('name', 'like', "%$search_term%");
```

Comment:
Searching for `tags` with their name attribute would be inefficient. Let's search it with the `key` attribute instead.

```php
if ($search_term) {
    $query
        ->where('key', 'like', '%' . Tag::constructKey($search_term) . '%');
}
```